omnigibson/object_states/particle_modifier.py

```diff
@@ -378,14 +349,13 @@ def _generate_limit_condition(self, system):
             function: Limit checker function, with signature condition(obj) --> bool, where @obj is the specific object
                 that this ParticleModifier state belongs to
         """
+        assert system in self.modified_particle_count, f"System {system.name} is not defined in the conditions."
```

Comment:
maybe more semantically clear to do `system in self.conditions`?

Reply:
done!